lib/danger/ci_source/gitlab.rb

```diff
@@ -0,0 +1,28 @@
+# https://circleci.com/docs/environment-variables
+require "uri"
+
+module Danger
+  class GitlabCI < CI
+    def self.validates_as_ci?(env)
+      env.key? "DRONE_REPO"
+    end
+
+    def self.validates_as_pr?(env)
+      env["DRONE_PULL_REQUEST"].to_i > 0
+    end
+
+    def self.validates?(env)
+      return !env["GITLAB_PROJECT_ID"].nil? && !env["GITLAB_MR_ID"].nil?
+    end
+
```

Comment:
These `validates_as` won't be hooked up correctly, they'll need porting to `self.validates_as_ci?` and `self.validates_as_pr?`

The first one is when you're sure that it's GitLabCI, the second is when you're sure it's actually a PR/MR